### ISSUE TITLE
Number of positives off

### ISSUE THREAD

**Issue by lucventurini** [bbuchfink/diamond#336]
Specific to the latest version (0.9.31). For example:

```xml
      <Hsp_num>1</Hsp_num>
      <Hsp_bit-score>70.9</Hsp_bit-score>
      <Hsp_score>172</Hsp_score>
      <Hsp_evalue>1.6e-12</Hsp_evalue>
      <Hsp_query-from>482</Hsp_query-from>
      <Hsp_query-to>742</Hsp_query-to>
      <Hsp_hit-from>8</Hsp_hit-from>
      <Hsp_hit-to>95</Hsp_hit-to>
      <Hsp_query-frame>2</Hsp_query-frame>
      <Hsp_hit-frame>0</Hsp_hit-frame>
      <Hsp_identity>39</Hsp_identity>
      <Hsp_positive>57</Hsp_positive>
      <Hsp_gaps>1</Hsp_gaps>
      <Hsp_align-len>88</Hsp_align-len>
         <Hsp_qseq>KTIQTNIRS-QYSSDLSSYASACKKYSSLKSFDSLLHERTNSIISSLAAQAKTRSLNIESLMEVYGYLLELNQDTVRVIIESKEDVLK</Hsp_qseq>
         <Hsp_hseq>KSQETSAKSVHYTTELRSYAAACKADTELQSFDTCLQARTSHVISTLATGVEVRALSFDSLKEVTQCLLEMNQEVVKVILDCKKDIWK</Hsp_hseq>
      <Hsp_midline>K+ +T+ +S  Y+++L SYA+ACK  + L+SFD+ L  RT+ +IS+LA   + R+L+ +SL EV   LLE+NQ+ V+VI++ K+D+ K</Hsp_midline>
    </Hsp>
```

The XML reports only 57 positives hits, while there are 39 identities and *25* positive hits, as attested by the midline.
The tabular output is off as well, as the `ppos` value (second to last) should be 39 + 25 = 64 / 88 = 72.7%, not 64.8%.

```
tr_c137_g1_i1.mrna1.77  Q9SYZ7  44.3    88      48      1       482     742     8       95      1.6e-12 70.9    64.8    1TSIQQE1NSIARK1-VQH1STSTDE1SR3SA3KAYDSTSE1KQ3STLC1HQEA2NSSHIV2ST2ATQGAVKETV1SA1NSIFED2MK2YTGQYC3LM2DETV1RK2ILEDSC1EK1VILW1
```

This happens only with the latest version, not with 0.9.30.

**Comment by lucventurini** [comment]
Query:
```
>tr_c137_g1_i1.mrna1.77
TCTCATGCCCATTAATAAATTGAGTTATCCTATATTTGCACAACATGGAGCATGATAGGA
AAAGATAAATCAATGATTAAAGAGAGTAATTAAACTTCTTGATCAAGGTGCGTTGATAGT
TTGATTTTGACATATCAACAAACTCCACAAAAATCAGACTACGGAAAAAAGAAATCATTT
CTTATGACAGCTTGTTCCACAGAAAAACATTAAAAAAGACCAAATCATATTACGTGCCCA
TTTTCGAGGTTTGACTAATTGTGGATTTTGGGGCGTTAAACCTTAGAACAAGAGGCTATA
TATATTAGTCAATCGTCAGAGACAGAAGTAATTAGCTCTTCCTGTGTCTTGGATTACAAT
TTCAAACGTTCCGGTCAGATTAGCTCCACTAGTAATATCAATGGTGTTCTGTGGATTATT
TTCGGAGCTCATAAAAGGACACTCCAGTTCTAACAACAACGGAACTAACGTTTCACTGGC
TAAAACGATCCAAACAAATATTCGGTCCCAATACTCATCAGACCTGAGTTCTTACGCATC
CGCTTGTAAAAAATACTCATCCCTCAAATCTTTTGATTCATTGCTTCACGAGCGAACAAA
CAGCATCATCAGCTCGCTCGCTGCTCAGGCCAAGACTCGATCTTTAAACATAGAGTCACT
CATGGAGGTCTACGGATATCTTCTTGAACTAAACCAAGATACAGTAAGAGTCATCATTGA
AAGCAAAGAAGATGTTTTGAAGA
```

Target:
```
>Q9SYZ7
MGNQTSKKSQETSAKSVHYTTELRSYAAACKADTELQSFDTCLQARTSHVISTLATGVEV
RALSFDSLKEVTQCLLEMNQEVVKVILDCKKDIWKNQEMFELVEDYFENSLKTLDFCAAL
EKGLRRARDSHLLILVALQQFEDESLVQGGNGYKKTLEELKNFKDAESPFNEDFFKMFQS
VYKQQMLMLEKLQHRKNKLDKKLKCIHTWRKLSSIIFVATFATVLICSVVAAAMAAPPVA
AALAAATAVPLGSMGKWIDSLWKNYENALKGQKEVISSMQAGTFVAVKDLDNIRVLIERL
EIEITGMVKSAEFAVEHNAVKIGIDDIKKKLEVFKKNVEELGTQADLCSRDIRRARTVIL
QRIIKHPNNASSST
```

**Comment by bbuchfink** [comment]
This should be fixed in f2e307bd58ad4bca6be0755f2da5eca1dc38e1a9.

**Comment by lucventurini** [comment]
Many thanks!
